lib/lookup-table/include/LookupTable.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#include <iterator>
+#include <optional>
+#include <unordered_map>
+
+// This a LUT type. It is not meant to be mutated after initilization.
+template <class Key, class Value, class Hash = std::hash<Key>, class KeyEqual = std::equal_to<Key>,
+          class Allocator = std::allocator<std::pair<const Key, Value>>>
+class LookupTable {
+ private:
+  const std::unordered_map<Key, Value, Hash, KeyEqual, Allocator> _unordered_map;
+
+ public:
+  using key_type        = Key;
+  using mapped_type     = Value;
+  using value_type      = std::pair<const Key, Value>;
+  using size_type       = size_t;
+  using difference_type = std::ptrdiff_t;
+  using hasher          = Hash;
+  using key_equal       = KeyEqual;
+  using allocator_type  = Allocator;
+  using reference       = value_type &;
+  using const_reference = const value_type &;
+  using pointer         = std::allocator_traits<Allocator>::pointer;
+  using const_pointer   = std::allocator_traits<Allocator>::const_pointer;
+
+  LookupTable() = delete;
+
+  // range based constructor
+  template <typename InputIterator>
+  LookupTable(InputIterator first, InputIterator last) : _unordered_map(first, last) {}
+  // copy constructor and assignment
+  LookupTable(const LookupTable &copy) = default;
+  LookupTable &operator=(const LookupTable &) = default;
+  // move constructor and assignment
+  LookupTable(LookupTable &&move) = default;
+  LookupTable &operator=(LookupTable &&) = default;
+  // initializer list constructor
+  LookupTable(std::initializer_list<value_type> init, size_type bucket_count = 0, const Hash &hash = Hash(),
+              const key_equal &equal = key_equal(), const Allocator &alloc = Allocator())
+      : _unordered_map(init, bucket_count, hash, equal, alloc) {}
+
+  typename std::unordered_map<Key, Value>::iterator begin() {
+    return _unordered_map.begin();
+  }
+  typename std::unordered_map<Key, Value>::iterator end() {
+    return _unordered_map.end();
+  }
+
+  std::size_t size() const {
+    return _unordered_map.size();
+  }
+  std::size_t max_size() const {
+    return _unordered_map.max_size();
+  }
+
+  bool empty() const {
+    return _unordered_map.empty();
+  }
+
+  std::optional<Value> at(const Key &key) const {
+    if (_unordered_map.find(key) == _unordered_map.end()) return std::nullopt;
```

Comment:
I personally think its bad style to have same line returns like this. Its less visible and more prone to error.

In the ros repo, i enforced always having braces and no single line ifs like this. 

In embedded systems I program for like work, typically every function only has a single return statement allowed. This is a MISRA requirement (automotive safety standard) and is basically used in all safety critical things in the embedded world to prevent unintentional bugs. Idk if its worth enforcing for the firmware team.

at the least i think this should be braced and on its own line

Reply:
i see. ill start looking to just have 1 return per function but i dont think we should enforce it in the repo

Reply:
i think thats fine, but still think we should do the brace thing

Reply:
u wanna enforce the brace thing? i kinda prefer not having the braces as it saves a line of code. i think we should leave this as a preference thing. was the brace rule enforced at ur coops?

Reply:
yeah the brace thing was always enforced. it makes branching operations a lot clearer and prevents the bug where u think you're adding to a branch but its actually a whitespace error and the statement is outside the branch